addons/mail/static/tests/message/message.test.js

```diff
@@ -1217,33 +1218,99 @@ test("Editing a message to clear its composer opens message delete dialog.", asy
     });
 });
 
-test("Clear message body should not open message delete dialog if it has attachments", async () => {
+test("[text composer] Clear message body should not open message delete dialog if it has attachments", async () => {
     const pyEnv = await startServer();
     const channelId = pyEnv["discuss.channel"].create({
         name: "general",
         channel_type: "channel",
     });
-    pyEnv["mail.message"].create({
-        author_id: serverState.partnerId,
-        body: "not empty",
-        model: "discuss.channel",
-        res_id: channelId,
-        message_type: "comment",
-        attachment_ids: [
-            pyEnv["ir.attachment"].create({ name: "test.txt", mimetype: "text/plain" }),
-        ],
-    });
+    pyEnv["mail.message"].create([
+        {
+            author_id: serverState.partnerId,
+            body: "no attachment",
+            model: "discuss.channel",
+            res_id: channelId,
+            message_type: "comment",
+        },
+        {
+            author_id: serverState.partnerId,
+            body: "not empty",
+            model: "discuss.channel",
+            res_id: channelId,
+            message_type: "comment",
+            attachment_ids: [
+                pyEnv["ir.attachment"].create({ name: "test.txt", mimetype: "text/plain" }),
+            ],
+        },
+    ]);
     await start();
     await openDiscuss(channelId);
-    await click(".o-mail-Message [title='Edit']");
+    await click(".o-mail-Message:eq(0) [title='Edit']");
     await insertText(".o-mail-Message.o-editing .o-mail-Composer-input", "", { replace: true });
     triggerHotkey("Enter");
-    await contains(".o-mail-Message-textContent", { text: "" });
-    // weak test, no guarantee that we waited long enough for the potential dialog to show
     await contains(".modal-body p", {
         text: "Are you sure you want to bid farewell to this message forever?",
-        count: 0,
     });
+    await click(".modal-footer button:contains('Delete')");
+    await contains(".o-mail-Message.o-editing .o-mail-Composer-input", { count: 0 });
+    await click(".o-mail-Message:eq(1) [title='Edit']");
+    await insertText(".o-mail-Message.o-editing .o-mail-Composer-input", "", { replace: true });
+    triggerHotkey("Enter");
+    await contains(".o-mail-Message.o-editing .o-mail-Composer-input", { count: 0 });
```

Comment:
Even all tests in this file are passing with reverting the fix, I think the fix is not covered by any test currently.